### DISCUSSION TITLE
Retrieving Instance ID token or FIS auth token for setting up A/B testing device

### DISCUSSION THREAD

**Discussion by lpatino10** [invertase/react-native-firebase#5364] [Q&A] [answered]
Hi everyone!

I've been using this library for a while to work with Remote Config and for setting up A/B tests. I recently found out that you can add a test device to an experiment to programmatically target variants as a sanity check before starting the experiment. See [here](https://firebase.google.com/docs/ab-testing/abtest-config?authuser=0#validate_your_experiment_on_a_test_device) for reference.

The Firebase UI is asking for an "Instance ID token or FIS auth token". I've tried pasting in the value I get from all of the following API methods, but in every case Firebase is telling me that the token is invalid:
- `analytics().getAppInstanceId()`
- `app.options.appId`
- `app.options.apiKey`
- `app.options.gaTrackingId`
- `app.options.messagingSenderId`

Does anyone know what the right method might be, or if this is just unsupported in this library? For reference, I'm using SDK version 11.4.1 and I'm trying to test this on the Android emulator.

Thanks!

**Comment by mikehardy** [comment, marked as answer]
I *think* what you want is `messaging.getToken()` ?

**Reply by lpatino10** [reply]
It looks like you were spot-on 😄  Thanks so much for the help and the quick reply!
